sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -164,6 +166,8 @@ def __init__(
             except TypeError:
                 self.snapshot = snapshot or path_snapshot
 
+        # This parameter is used for the hierarchy traversal. Give precedence to credential.
+        self.client_credential = credential if credential else sas_token
```

Comment:
why don't we use self.credential on base_client?

Reply:
Because it fails in the following case:
1. User creates a blob client with a SAS token (`BlobClient(.., credential=sas,..)`)
2. Eventually the method `self._format_query_string(sas_token, credential, snapshot=self.snapshot)` is called
3. This method would set `credential=None` and add the sas token to the URI/query parameters
4. So when we call `self.credential` we would get None which would create a ContainerClient with no credential

Reply:
My concern is user will see self.credential and self.client_credential on the same object

Reply:
probably make it self._raw_credential?

Reply:
probably we can keep raw url, which probably has the sas, while later we can just pass the raw url back to ContainerClient

Reply:
Comment on this below